runtime/src/status_cache.rs

```diff
@@ -183,16 +183,27 @@ impl<T: Serialize + Clone> StatusCache<T> {
         res: T,
     ) {
         let max_key_index = key.as_ref().len().saturating_sub(CACHED_KEY_SIZE + 1);
+
+        // Get the cache entry for this blockhash.
         let hash_map = self.cache.entry(*transaction_blockhash).or_insert_with(|| {
             let key_index = thread_rng().gen_range(0..max_key_index + 1);
             (slot, key_index, HashMap::new())
         });
 
+        // Update the max slot observed to contain txs using this blockhash.
         hash_map.0 = std::cmp::max(slot, hash_map.0);
+
+        // Grab the key slice.
         let key_index = hash_map.1.min(max_key_index);
         let mut key_slice = [0u8; CACHED_KEY_SIZE];
         key_slice.clone_from_slice(&key.as_ref()[key_index..key_index + CACHED_KEY_SIZE]);
-        self.insert_with_slice(transaction_blockhash, slot, key_index, key_slice, res);
+
+        // Insert the slot and tx result into the cache entry associated with
+        // this blockhash and keyslice.
+        let forks = hash_map.2.entry(key_slice).or_default();
+        forks.push((slot, res.clone()));
```

Comment:
These 2 lines were originally performed inside of `fn insert_with_slice`, so had to pull them up here